napi/scripts/napi.js

```diff
@@ -24,12 +24,30 @@ const moduleName = path.basename(process.cwd());
 process.env.NODE_INCLUDE_PATH = nodeIncludePath
 process.env.NODE_MAJOR_VERSION = nodeMajorVersion
 
+const platform = os.platform()
+var libExt = ''
+
+switch(platform) {
+	case 'darwin':
+		libExt = '.dylib'
+		break;
+	case 'win32':
+		libExt = '.dll'
+		break;
+	case 'linux':
+		libExt = '.so'
+		break;
+	default:
+		console.error('Operating system not currently supported or recognized by the build script')
+		process.exit(1)
+}
+
 switch (subcommand) {
   case 'build':
     const releaseFlag = argv.release ? '--release' : ''
     const targetDir = argv.release ? 'release' : 'debug'
-    cp.execSync(`cargo rustc ${releaseFlag} -- -Clink-args=\"-undefined dynamic_lookup -export_dynamic\"`, {stdio: 'inherit'})
-    cp.execSync(`cp target/${targetDir}/{lib${moduleName}.dylib,${moduleName}.node}`, {stdio: 'inherit'})
+    cp.execSync(`cargo rustc ${releaseFlag} -- -Clink-args=\"-undefined=dynamic_lookup -export_dynamic\"`, {stdio: 'inherit'})
```

Comment:
For me, the `=` sign in `-undefined=dynamic_lookup` is causing an error. It only works when we go back to replacing `=` with a space.

Reply:
That must be a mac/linux thing because I can only compile using Clang 3.9 on Ubuntu 16.04 LTS with the '=' sign. I will remove the package lock changes and add the '=' sign to the os case statement.

Reply:
Does it work for you with `-undefined,dynamic_lookup` as that also works with Linux.

Reply:
The problem seems to be that on MacOS, `linker -flag <arg>' is valid but on Linux it is being seen as `linker -flag [linker-file-target]`. If `linker -flag,<arg>' (no space) works for MacOS then this seems like a cross-platform fix. If anyone has more insight on the discrepancy please let me know.